include/numerics/type_vector.h

```diff
@@ -1100,6 +1100,31 @@ T cross_norm(const TypeVector<T> & b,
   return std::sqrt(cross_norm_sq(b,c));
 }
 
+template <typename T>
+inline
+TypeVector<T> TypeVector<T>::unit() const
+{
+
+  auto && length = norm();
+
+  libmesh_assert_not_equal_to (length, static_cast<Real>(0.));
```

Comment:
Does this line still work if `length` is non-`Real`?

Reply:
This works for all `DualNumber` types at least because we overload the comparison operators.